check.go

```diff
@@ -3,13 +3,14 @@ package mtree
 import (
 	"fmt"
 	"os"
-	"path/filepath"
 	"sort"
 )
 
 // Result of a Check
 type Result struct {
 	Failures []Failure // list of any failures in the Check
+	Missing  []Entry
+	Extra    []Entry
```

Comment:
I prefer the way I did it in #34 (it's more extensible). Also, I think we should merge #34 first so that this can take advantage of the interfaces.